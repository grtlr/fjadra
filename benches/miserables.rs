use criterion::{black_box, criterion_group, criterion_main, Criterion};
use fjadra::{Center, Link, ManyBody, Node, SimulationBuilder};
use fjadra_data::miserables;

fn bench_simulation(c: &mut Criterion) {
    let graph = miserables::Graph::load().unwrap();

    c.bench_function("miserables (init)", |b| {
        b.iter(|| {
            let simulation = SimulationBuilder::default()
                .build(graph.nodes.iter().map(|_| Node::default()))
                .add_force(
                    "link",
                    Link::new(graph.links.iter().map(|link| (link.source, link.target))),
                )
                .add_force("charge", ManyBody::new())
                .add_force("center", Center::new());

            black_box(simulation)
        });
    });

    c.bench_function("miserables (full)", |b| {
        b.iter(|| {
            let mut simulation = SimulationBuilder::default()
                .build(graph.nodes.iter().map(|_| Node::default()))
                .add_force(
                    "link",
                    Link::new(graph.links.iter().map(|link| (link.source, link.target))),
                )
                .add_force("charge", ManyBody::new())
                .add_force("center", Center::new());

            simulation.iter().last()
        });
    });
}

criterion_group!(benches, bench_simulation);
criterion_main!(benches);
