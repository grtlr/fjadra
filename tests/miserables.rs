use fjadra::{Center, Link, ManyBody, Node, SimulationBuilder};
use fjadra_data::miserables;

#[test]
fn test_miserables() {
    let graph = miserables::Graph::load().unwrap();

    let mut simulation = SimulationBuilder::default()
        .build(graph.nodes.iter().map(|_| Node::default()))
        .add_force(
            "link",
            Link::new(graph.links.iter().map(|link| (link.source, link.target))),
        )
        .add_force("charge", ManyBody::new())
        .add_force("center", Center::new());

    let positions = simulation.iter().last();

    insta::assert_debug_snapshot!(positions);
}
