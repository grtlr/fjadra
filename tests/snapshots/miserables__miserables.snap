---
source: tests/miserables.rs
assertion_line: 19
expression: positions
snapshot_kind: text
---
[
  [
    58.10029,
    -77.99321
  ],
  [
    63.59271,
    -118.67026
  ],
  [
    -3.01944,
    -42.86134
  ],
  [
    7.20054,
    -24.92717
  ],
  [
    83.63348,
    -109.11966
  ],
  [
    95.35948,
    -95.23362
  ],
  [
    98.97485,
    -73.50421
  ],
  [
    94.67002,
    -106.27625
  ],
  [
    101.57624,
    -85.91146
  ],
  [
    76.86627,
    -117.81671
  ],
  [
    -69.74987,
    56.00774
  ],
  [
    -50.20784,
    20.26441
  ],
  [
    -88.02864,
    -55.14069
  ],
  [
    -78.54806,
    48.16673
  ],
  [
    -84.65419,
    37.12759
  ],
  [
    -56.90253,
    58.44487
  ],
  [
    -92.42244,
    -157.83979
  ],
  [
    -155.08235,
    -180.42602
  ],
  [
    -141.0393,
    -220.82944
  ],
  [
    -136.48566,
    -192.50361
  ],
  [
    -159.92772,
    -212.18748
  ],
  [
    -123.33109,
    -209.89383
  ],
  [
    -170.17657,
    -193.74759
  ],
  [
    -114.2067,
    -114.63919
  ],
  [
    -21.81005,
    2.13962
  ],
  [
    13.1735,
    26.5494
  ],
  [
    -18.58546,
    -55.49706
  ],
  [
    -38.83046,
    12.67908
  ],
  [
    -102.5384,
    -4.71701
  ],
  [
    -134.80843,
    2.23234
  ],
  [
    -140.65518,
    -90.97171
  ],
  [
    -98.7993,
    -46.71075
  ],
  [
    -87.04401,
    18.37951
  ],
  [
    -77.3519,
    -2.18729
  ],
  [
    -143.92009,
    33.81547
  ],
  [
    -164.44119,
    33.05311
  ],
  [
    -130.01125,
    52.82729
  ],
  [
    -139.56973,
    69.61238
  ],
  [
    -158.63421,
    56.6518
  ],
  [
    57.31143,
    -9.95849
  ],
  [
    3.12316,
    -2.45565
  ],
  [
    51.91398,
    65.7349
  ],
  [
    4.49818,
    47.92083
  ],
  [
    -59.65291,
    -27.99714
  ],
  [
    -101.83831,
    27.13983
  ],
  [
    -145.87264,
    -20.1419
  ],
  [
    88.79385,
    207.75924
  ],
  [
    84.70377,
    164.37007
  ],
  [
    80.70999,
    102.95377
  ],
  [
    26.41722,
    -21.64094
  ],
  [
    -16.24997,
    -39.68912
  ],
  [
    19.71857,
    -51.67751
  ],
  [
    50.7443,
    -66.60096
  ],
  [
    15.34942,
    -95.662
  ],
  [
    52.31365,
    -39.11355
  ],
  [
    81.47592,
    22.53438
  ],
  [
    77.43258,
    -15.77648
  ],
  [
    142.30111,
    59.9559
  ],
  [
    99.95098,
    77.90252
  ],
  [
    143.46632,
    108.89338
  ],
  [
    166.24459,
    125.70861
  ],
  [
    154.69974,
    84.49315
  ],
  [
    135.48008,
    89.61423
  ],
  [
    170.30652,
    103.29997
  ],
  [
    112.16662,
    97.04123
  ],
  [
    172.20348,
    84.74532
  ],
  [
    155.24929,
    141.20793
  ],
  [
    182.20604,
    49.26465
  ],
  [
    -3.84195,
    67.13733
  ],
  [
    -14.90013,
    81.2619
  ],
  [
    17.29145,
    59.47571
  ],
  [
    3.94833,
    87.05536
  ],
  [
    -48.18818,
    -26.9848
  ],
  [
    96.72786,
    146.90546
  ],
  [
    67.49183,
    148.2767
  ],
  [
    27.90659,
    101.10221
  ],
  [
    136.13446,
    127.78192
  ]
]
